SPEC.md

```diff
@@ -442,17 +460,22 @@ Note that for the purpose of new lines, CRLF is considered _a single newline_.
 ## Full Grammar
 
 ```
-nodes := linespace* (node nodes?)? linespace*
+nodes := (line-space* node)* line-space*
+
+plain-line-space := newline | ws | single-line-comment
+plain-node-space := ws* escline ws* | ws+
 
-node := ('/-' node-space*)? type? identifier (node-space+ node-prop-or-arg)* (node-space* node-children ws*)? node-space* node-terminator
-node-prop-or-arg := ('/-' node-space*)? (prop | value)
-node-children := ('/-' node-space*)? '{' nodes '}'
-node-space := ws* escline ws* | ws+
+line-space := (plain-line-space+ '/-' plain-node-space* node)* plain-line-space+
+node-space := (plain-node-space+ '/-' plain-node-space* (node-prop-or-arg | node-children))* plain-node-space+
+
+node := type? identifier (node-space+ node-prop-or-arg)* (node-space+ node-children)? node-space* node-terminator
```

Comment:
One remaining spec bug we should iron out: currently

```kdl
node /-"argument";
```

isn't valid, as `node-space` requires trailing `plain-node-space`. The best fix I can think of is to take the trailing `plain-node-space` out of `node-space` and put it into `node`, e.g.

```suggestion
line-space := plain-line-space+ ('/-' plain-node-space* node)?
node-space := plain-node-space+ ('/-' plain-node-space* (node-prop-or-arg | node-children))?

node := type? identifier (node-space* plain-node-space+ node-prop-or-arg)* (node-space* plain-node-space+ node-children)? node-space* node-terminator
```

Thoughts? Checks that we're speccing what we want to be?

Reply:
An alternative formulation, perhaps better:

```suggestion
line-space := plain-line-space+ ('/-' plain-node-space* node)?
node-space := plain-node-space+ ('/-' plain-node-space* (node-prop-or-arg | node-children))?

required-node-space := node-space* plain-node-space+
optional-node-space := node-space*

node := type? identifier (required-node-space node-prop-or-arg)* (required-node-space node-children)? optional-node-space node-terminator
```